code/modules/reagents/reagent_containers/food/drinks.dm

```diff
@@ -11,9 +23,27 @@
 	amount_per_transfer_from_this = 5
 	volume = 50
 	var/shaken = 0
+	var/drink_flags = NO_EMPTY_ICON
+	var/empty_icon_state = null
+
+/obj/item/reagent_containers/food/drinks/Initialize()
+	. = ..()
+	if(drink_flags & IS_GLASS)
+		unacidable = TRUE
 
 /obj/item/reagent_containers/food/drinks/on_reagent_change()
-	return
+	update_icon()
+
+/obj/item/reagent_containers/food/drinks/update_icon()
+	if(!reagents.total_volume)
+		if(drink_flags & UNIQUE_EMPTY_ICON)
```

Comment:
```suggestion
		if(drink_flags & empty_icon)
```

You don't need to put in a new flag here. Just check whether if it has an empty icon, and then proceed from there.

Reply:
I suppose I could, though the format suggested would not work. I would just do if(empty_icon_state).

Reply:
Alright, then go ahead with that alternative